acra-censor/acra-censor_test.go

```diff
@@ -1070,30 +1090,78 @@ func TestSerialization(t *testing.T) {
 		t.Fatal(err)
 	}
 }
-func TestLogging(t *testing.T) {
+func TestSerializationOnSameQueries(t *testing.T) {
+	// 5 queries, 3 unique redacted queries
+	numOfUniqueQueries := 3
+	testQueries := []string{
+		// will be redacted
+		"SELECT NAME WHERE EMP_ID = '1234';",
+		"SELECT NAME WHERE EMP_ID = '345';",
+
+		// different
+		"SELECT EMP_ID, LAST_NAME FROM EMPLOYEE WHERE CITY = 'Seattle' ORDER BY EMP_ID;",
+		"SELECT EMP_ID FROM EMPLOYEE WHERE CITY = 'Seattle' ORDER BY EMP_ID;",
+
+		// similar to previous one, will be redacted
+		"SELECT EMP_ID FROM EMPLOYEE WHERE CITY = 'London' ORDER BY EMP_ID;",
+	}
+	tmpFile, err := ioutil.TempFile("", "censor_log")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err = tmpFile.Close(); err != nil {
+		t.Fatal(err)
+	}
+	handler, err := handlers.NewQueryCaptureHandler(tmpFile.Name())
+	defer handler.Release()
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, query := range testQueries {
+		_, err = handler.RedactAndCheckQuery(query)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	defaultTimeout := handler.GetSerializationTimeout()
+	handler.SetSerializationTimeout(50 * time.Millisecond)
+	//wait until goroutine handles complex serialization
+	time.Sleep(defaultTimeout + handler.GetSerializationTimeout() + 10*time.Millisecond)
+
+	if len(handler.GetAllInputQueries()) != numOfUniqueQueries {
+		t.Fatal("Expected to have " + fmt.Sprint(numOfUniqueQueries) + " unique queries. \n Got:" + strings.Join(handler.GetAllInputQueries(), " | "))
+	}
+	err = handler.DumpAllQueriesToFile()
+	if err != nil {
+		t.Fatal(err)
+	}
+	handler.Reset()
+	if len(handler.GetAllInputQueries()) != 0 {
+		t.Fatal("Expected no queries \nGot: " + strings.Join(handler.GetAllInputQueries(), " | "))
+	}
+	err = handler.ReadAllQueriesFromFile()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(handler.GetAllInputQueries()) != numOfUniqueQueries {
+		t.Fatal("Expected to have " + fmt.Sprint(numOfUniqueQueries) + " unique queries. \n Got:" + strings.Join(handler.GetAllInputQueries(), " | "))
+	}
+	for index, query := range handler.GetAllInputQueries() {
+		if strings.EqualFold(testQueries[index], query) {
+			t.Fatal("Expected: " + testQueries[index] + "\nGot: " + query)
+		}
+	}
+	if err = os.Remove(tmpFile.Name()); err != nil {
```

Comment:
better to defer file removing after creation because it will exists after any `t.Fatal` call above

Reply:
totally makes sense, added everywhere